ios/Breez Notification Service Extension/NotificationService.swift

```diff
@@ -87,41 +75,51 @@ class NotificationService: UNNotificationServiceExtension {
     }
     
     private func shutdown() -> Void {
-        self.logger.info("shutding down with \(self.paymentReceivers.count) tasks")
-        for r in self.paymentReceivers {
-            r.displayPushNotification(title: "Receive payment failed")
-        }
-        self.paymentReceivers = []
-        self.paymentHashPollerTimer?.invalidate()
-    }
-    
-    private func onPaymentReceived(payment: Payment) -> Void {
-        self.logger.info("onPaymentReceived for \(payment.amountMsat) sats")
-        guard case .ln(let data) = payment.details else {
-            return
-        }
-        for r in self.paymentReceivers {
-            if r.paymentHash == data.paymentHash {
-                r.displayPushNotification(title: "Received \(payment.amountMsat/1000) sats")
-            }
-        }
-        self.paymentReceivers.removeAll(where: {$0.paymentHash == data.paymentHash})
+        Self.logger.info("shutting down...")
+        BreezManager.unregister()
+        Self.logger.info("task unregistered")
+        self.currentTask?.onShutdown()
     }
 }
 
-class PaymentReceiver {
+class PaymentReceiver : SDKBackgroundTask {
     private var contentHandler: ((UNNotificationContent) -> Void)?
     private var bestAttemptContent: UNMutableNotificationContent?
     private var logger: XCGLogger
-    public var paymentHash: String
+    private var receivedPayment: Payment? = nil
     
-    init(logger: XCGLogger, contentHandler: ((UNNotificationContent) -> Void)? = nil, bestAttemptContent: UNMutableNotificationContent? = nil, paymentHash: String) {
+    init(logger: XCGLogger, contentHandler: ((UNNotificationContent) -> Void)? = nil, bestAttemptContent: UNMutableNotificationContent? = nil) {
         self.contentHandler = contentHandler
         self.bestAttemptContent = bestAttemptContent
-        self.paymentHash = paymentHash
         self.logger = logger
     }
     
+    func onShutdown() {
+        if let p =  self.receivedPayment {
+            self.displayPushNotification(title: "Received \(p.amountMsat/1000) sats")
+        } else {
+            self.displayPushNotification(title: "Receive payment failed")
+        }
```

Comment:
I don't remember the exact syntax but could be reduced to sth like
```suggestion
        let title = self.receivedPayment.first { "Received \($0.amountMsat/1000) sats" } ??  "Receive payment failed"
        self.displayPushNotification(title: title)
```

Reply:
receivedPayment is not an array so no first so I changes it slightly different.